src/vs/workbench/contrib/tasks/common/problemMatcher.ts

```diff
@@ -220,6 +220,7 @@ export async function getResource(filename: string, matcher: ProblemMatcher, fil
 	if (fullPath[0] !== '/') {
 		fullPath = '/' + fullPath;
 	}
+	fullPath = normalize(fullPath);
```

Comment:
I'm a beginner on vscode's code, could you explain to me how does this line avoids this:
>Actual: A new editor is opened containing the relative part of the path. The file history contains a new entry. When using "Go to file", both file locations are offered.

?

Reply:
Without this change, two editors will be opened. One for `/Users/foo/dev/project/main.cpp` and one for `/Users/foo/dev/project/output/../main.cpp`, even though it refers to the same file. By normalizing it upfront, the seconds path will be identical to the first one.

Ideally, vscode normalizes the paths before opening an editor but that sounded like a bigger change to me.